### PR TITLE
Fishes love kronkaine + Examining fishing spots.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Food items with kronkaine in it now count as great baits (this is the case of the "all-natural" bait from that overpriced cargo pack I'm converting into a less expensive goodie in another PR)

Sufficiently high fishing skill (and the relative skillchip) allows you to list fishes in a fishing spot by examining it twice.

## Why It's Good For The Game
Little fishing things off the top of my head before I move on bigger stuff.

## Changelog

:cl:
add: Fishes love kronkaine.
qol: Examining a fishing spot twice with sufficiently high fishing skill (or the skillchip) will get you a list of fishes that can be caught.
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
youtube.com/watch?v=1029PjJJUnk

File: code/datums/components/fishing_spot.dm [hunk 1]
```diff
@@ -28,6 +33,43 @@
...
+		if(!ispath(reward, /obj/item/fish))
+			continue
+		var/obj/item/fish/prototype = reward
+		if(initial(prototype.show_in_catalog))
+			has_known_fishes = TRUE
+			break
+	if(!has_known_fishes)
+		return
+
+	. += span_tinynoticeital("This is a fishing spot. You can look again to list its fishes...")
```
Comment (san7890) [inline code/datums/components/fishing_spot.dm:53]:
```suggestion
	examine_text += span_tinynoticeital("This is a fishing spot. You can look again to list its fishes...")
```

Remember, `SIGNAL_HANDLER` procs (as in procs that we register signals to) should only return bitfields.

File: code/datums/components/fishing_spot.dm [hunk 2]
```diff
@@ -28,6 +33,43 @@
...
+		if(!ispath(reward, /obj/item/fish))
+			continue
+		var/obj/item/fish/prototype = reward
+		if(initial(prototype.show_in_catalog))
+			known_fishes += initial(prototype.name)
+
+	if(!length(known_fishes))
+		return
+
+	. += span_info("A list of fishes you can catch here: [english_list(known_fishes)].")
```
Comment (san7890) [inline code/datums/components/fishing_spot.dm:71]:
```suggestion
	examine_text += span_info("You can catch the following fish here: [english_list(known_fishes)].")
```

code fix and rewording the sentence to be less clunky

File: code/datums/skills/fishing.dm [hunk 3]
```diff
@@ -4,21 +4,26 @@
...
+	title = "Angler"
 	desc = "How empty and alone you are on this barren Earth."
 	modifiers = list(SKILL_VALUE_MODIFIER = list(1, 1, 0, -1, -2, -4, -6))
 	skill_item_path = /obj/item/clothing/head/soft/fishing_hat
 
 /datum/skill/fishing/New()
 	. = ..()
-	levelUpMessages[SKILL_LEVEL_MASTER] = span_nicegreen("After lots of practice, I've begun to truly understand the surprising depth behind [name]. As a master [title], I can take an easier guess of what I'm trying to catch now.")
+	levelUpMessages[SKILL_LEVEL_JOURNEYMAN] = span_nicegreen("I feel like I've become quite proficient at [name]! I can tell what fishes I can catch at any given fishing spot.")
+	levelUpMessages[SKILL_LEVEL_MASTER] = span_nicegreen("I've begun to truly understand the surprising depth behind [name]. As a master [title], I can soft of guess of what I'm going to catch (instead of plain \"!!!\").")
```
Comment (san7890) [inline code/datums/skills/fishing.dm:15]:
```suggestion
	levelUpMessages[SKILL_LEVEL_MASTER] = span_nicegreen("I've begun to truly understand the surprising depth behind [name]. As a master [title], I can guess what I'm going to catch now!")
```

I think it's pretty clunky to reference the `!!!` message here, as well as cleaning up the insecurities

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 4]
```diff
@@ -785,6 +785,23 @@
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 	addiction_types = list(/datum/addiction/stimulants = 20)
 
+/datum/reagent/drug/kronkaine/on_new(data)
+	. = ..()
+	// Kronkaine also makes for a great fishing bait (found in "natural" baits)
+	if(istype(holder?.my_atom, /obj/item/food))
```
Comment (san7890) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm:791]: early return here?

**Review by san7890 — CHANGES_REQUESTED** [review]
nits, but also some really serious code considerations.

**Comment by xXPawnStarrXx** [comment]
I adore this, I spent ages trying to use the book to find where certain fish spawned and remember. Having the option to get the chip is so much better. Does this only show fish or 'dud' items too? I'd rather not see 'duds'.

I forget, is the skillchip available via cargo?

**Comment by Ghommie** [comment]
> I adore this, I spent ages trying to use the book to find where certain fish spawned and remember. Having the option to get the chip is so much better. Does this only show fish or 'dud' items too? I'd rather not see 'duds'.
> 
> I forget, is the skillchip available via cargo?

Only fish, also the library.

But hold your horses for a day, 'cuz I omitted to register a signal so examining fishing spots doesn't really work rn. :P

**Comment by xXPawnStarrXx** [comment]
> But hold your horses for a day, 'cuz I omitted to register a signal so examining fishing spots doesn't really work rn. :P

I mostly play on a downstream, so these PRs take like six weeks to get to us, I got nothing but patience at this point.

**Pull request merged**